extensions/amp-megaphone/0.1/amp-megaphone.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Megaphone podcast
+ *
+ * Example:
+ * <code>
+ * <amp-megaphone
+ *   height=166
+ *   data-episode="DEM6617440160"
+ *   data-light="true"
+ *   layout="fixed-height">
+ * </amp-megaphone>
+ */
+
+import {Layout} from '../../../src/layout';
+import {addParamsToUrl} from '../../../src/url';
+import {dict} from '../../../src/utils/object';
+import {getData, listen} from '../../../src/event-helper';
+import {isObject} from '../../../src/types';
+import {removeElement} from '../../../src/dom';
+import {startsWith} from '../../../src/string';
+import {tryParseJson} from '../../../src/json';
+import {userAssert} from '../../../src/log';
+
+class AmpMegaphone extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {boolean} */
+    this.isPlaylist_ = false;
+
+    /** @private {string} */
+    this.baseUrl_ = '';
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url(this.baseUrl_, opt_onLayout);
+    this.preconnect.url('https://assets.megaphone.fm', opt_onLayout);
+    this.preconnect.url('https://megaphone.imgix.net', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return (
```

Comment:
Create an `isLayoutFixedSize` next to `isLayoutSizeDefined`, and make it the latter use the former.